metricbeat/mb/testing/modules.go

```diff
@@ -262,17 +274,16 @@ func RunPushMetricSetV2(timeout time.Duration, waitEvents int, metricSet mb.Push
 	}()
 
 	timeoutC := time.After(timeout)
-	for {
+	for done := false; !done; {
 		select {
 		case <-termC:
-			return r.Events
+			done = true
 		case <-timeoutC:
-			return r.Events
+			done = true
 		default:
-			if waitEvents != 0 && waitEvents <= r.eventCount() {
-				return r.Events
-			}
 			time.Sleep(100 * time.Millisecond)
```

Comment:
@andrewkroh What was the reason we wait here 100ms? Perhaps worth a comment (not related to this PR).

Reply:
I added this sleep in a previous commit, to avoid constant checking of the exit condition until the `termC` or `timeoutC` channels fire. Again, this is just a testing helper, but I can add an explanatory comment if you think it's necessary.